x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTask.java

```diff
@@ -185,9 +210,30 @@ private void sendShardChangesRequest(long from, int maxOperationCount, long maxR
     }
 
     private void sendShardChangesRequest(long from, int maxOperationCount, long maxRequiredSeqNo, AtomicInteger retryCounter) {
+        final long startTime = relativeTimeProvider.getAsLong();
         innerSendShardChangesRequest(from, maxOperationCount,
-            response -> handleReadResponse(from, maxRequiredSeqNo, response),
-            e -> handleFailure(e, retryCounter, () -> sendShardChangesRequest(from, maxOperationCount, maxRequiredSeqNo, retryCounter)));
+                response -> {
+                    synchronized (ShardFollowNodeTask.this) {
+                        // noinspection NonAtomicOperationOnVolatileField
```

Comment:
see comment about these not needing to be volatile, meaning we can get rid of this suppression.